### ISSUE TITLE
RGroupDecomposition GA hangs

### ISSUE THREAD

**Issue by bp-kelley** [rdkit/rdkit#4106]
**Describe the bug**
The RGroup decomposition hangs with a small input set when using the Genetic algorithm.  Using the original scoring system doesn't have the issue.

**To Reproduce**
```
from rdkit.Chem import rdRGroupDecomposition as rgd
from rdkit import Chem

decomp_params = rgd.RGroupDecompositionParameters()
decomp_params.matchingStrategy = rgd.GA
decomp_params.onlyMatchAtRGroups = False
decomp_params.scoreMethod = rgd.RGroupScore.FingerprintVariance

scaffold = Chem.MolFromSmiles("CC.C(N)C")
mols = [Chem.MolFromSmiles(m) for m in ('NCCOC(C)C', 'PCCONC(N)C', 'SCCOPSCCCl')]

decomposer = rgd.RGroupDecomposition(scaffold, decomp_params)
for mol in mols:
    decomposer.Add(mol)
decomposer.Process()
```

**Expected behavior**
Not hanging :)

**Screenshots**

**Configuration (please complete the following information):**
 - RDKit version: 2021.03
 - OS: OS/X

**Comment by bp-kelley** [comment]
@jones-gareth This might be in your court :)

**Comment by jones-gareth** [comment]
@bp-kelley 

This is fixed in master- see https://github.com/rdkit/rdkit/pull/3934 (approved by yourself!).  I don't think that fix made the cut for 2021.03.  I've confirmed that your example code does not hang on a current build from master (Ubuntu 18.04).

**Comment by bp-kelley** [comment]
Great!  Guess I need to update my builds :)
